src/renderers/WebGLRenderer.js

```diff
@@ -1982,6 +1982,12 @@ function WebGLRenderer( parameters ) {
 
 			WebGLUniforms.upload( _gl, materialProperties.uniformsList, m_uniforms, textures );
 
+			if ( material.isShaderMaterial ) {
+
+				material.uniformsNeedUpdate = false;
```

Comment:
@mrdoob @Mugen87 https://github.com/mrdoob/three.js/pull/15581#issuecomment-454833117 I think this, at a minimum, needs an explanatory comment.

Reply:
You mean something like `// #15581`? Or a textual description?

Reply:
I think `// #15581` is OK.